rescompy/rescompy.py

```diff
@@ -209,65 +208,126 @@ class TrainResult:
+			    the training period, including the initial transient, for the 
+                corresponding training task.
                 The first axis must have the same length as the first axis of
                 states.
+            target_outputs (list): The list of target_outputs stored during the
+			    training stage.
+			transient_lengths (list): Each entry contains the length of the 
+                initial transient for the corresponding training task.
+			accessible_drives (list): The list of training signal IDs that have
+			    been stored during training. The entries of drive_results and 
+				target_outputs corresponding to input signals that are not 
+				identified in accessible_drives will be None.
             feature_function (Callable): The function that transforms the
                 states and inputs to features.
             weights (np.ndarray): The array of output weights.
-            transient_length (int): The length of the initial transient.
         """
-        
-        # Check for shape consistencies.
-        features = feature_function(states, inputs)
-        utils.check_shape(inputs.shape, (states.shape[0], None),
-                          'inputs')
-        utils.check_shape(target_outputs.shape,
-                          (states.shape[0], None), 'target_outputs')
-        utils.check_shape(weights.shape, (features.shape[1],
-                                          target_outputs.shape[1]),
-                          'weights')
+		
+        #Check for shape consistencies.
+        for task_ind in range(len(inputs)):
+            features = feature_function(states[task_ind], inputs[task_ind])
+            utils.check_shape(inputs[task_ind].shape,
+							  (states[task_ind].shape[0], None), 'inputs')
+            utils.check_shape(weights.shape,
+							  (features.shape[1], target_outputs[0].shape[1]),
+							  'weights')
+            if (target_outputs[task_ind].shape[0] > 1):
+                utils.check_shape(target_outputs[task_ind].shape,
+								  (states[task_ind].shape[0], None), 'target_outputs')
+
+            else:
+                utils.check_range(features.shape[0],
+								  "The number of feature vectors per input signal",
+								  1, 'eq', True)
         
         # Assign attributes.
-        self.states = states
-        self.inputs = inputs
-        self.target_outputs = target_outputs
         self.feature_function = feature_function
+        self.accessible_drives = accessible_drives
         self.weights = weights
-        self.transient_length = transient_length
         self.jacobian = jacobian
         
+		# Check if the feature function requires a lookback period to include
+        # time-delayed states or inputs in feature vectors.
+        if hasattr(feature_function, 'lookback_length'):
+            self.lookback_length = feature_function.lookback_length
+        else: self.lookback_length = 0
+
+        self.listed_targets = target_outputs		
```

Comment:
what are these listed_* properties?

Reply:
self.listed_states stores the reservoir states from each training task as a separate entry in a list of arrays (one list entry/array per training task/input signal). self.states concatenates all the entries of self.listed_states in a single array, as in the current rescompy. The same is true for the other listed_* properties.